glean-core/src/core/mod.rs

```diff
@@ -687,6 +694,20 @@ impl Glean {
         metric.test_get_value(self)
     }
 
+    /// Set configuration for metrics' disabled property, typically from a remote_settings experiment
+    /// or rollout
+    ///
+    /// # Arguments
+    ///
+    /// * `json` - The stringified JSON representation of a `MetricsDisabledConfig` object
+    pub fn set_metrics_disabled_config(&self, cfg: MetricsDisabledConfig) {
+        // Set the current MetricsDisabled object
+        *self.remote_settings_metrics_config.lock().unwrap() = cfg;
+
+        // Update remote_settings epoch
+        self.remote_settings_epoch.fetch_add(1, Ordering::SeqCst);
```

Comment:
Would it be better to release the lock immediately after setting the `cfg`?

Reply:
That's what the current code does in that one-liner.
I argue it's more correct to keep it until the epoch is also incremented.
Otherwise _in theory_ someone can read the new config but the old epoch.